sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java

```diff
@@ -98,21 +98,28 @@
    *
    * @param fileOrPatternSpec {@link IOChannelFactory} specification of file or file pattern
    *        represented by the {@link FileBasedSource}.
+   * @param failOnEmptyInput whether to throw if {@code fileOrPatternSpec} does not match any
+   *        files.
    * @param minBundleSize minimum bundle size in bytes.
    */
-  public FileBasedSource(String fileOrPatternSpec, long minBundleSize) {
```

Comment:
I've tried to find and update all consumers of `FileBasedSource` and derived sources. The only usage I found which I didn't update is [`BigQuerySourceBase.createSources()`](https://github.com/apache/incubator-beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java#L1256). It uses `AvroSource`s internally to read temp files created from the export job. These are expected to exist and I believe we should fail if they are missing.